Add functionality for users to retrieve template by Name as well as UUID

mmlb opened the pull request:
## Description

This PR adds the ability for Tink to retrieve workflows by either UUID or by Name.
If the value after `tink template get` is not a valid UUID, Tink will default to searching for the value in the Name field of the templates.

Mostly developed by @Belchy06, thanks!
My additions are:
 - cleaned up the import blocks so its just 2 blocks
 - fix typo in help output
 - rebase on latest main
 - rework some of the logic in cmd/tink-cli/cmd/get/get.go to be clearer

## Why is this needed

This feature is required for a project @Belchy06 is working on but is also something I'd like to make use of.

## How Has This Been Tested?

This has been tested in the VM's based on the [Local Setup with Vagrant](https://docs.tinkerbell.org/setup/local-vagrant/) tutorial, as well as in a personal deployment by @Belchy06.

mmlb commented on cmd/tink-cli/cmd/get/get.go:
```diff
@@ -155,3 +148,31 @@ func NewGetCommand(opt Options) *cobra.Command {
 	return cmd
 }
+
+func retrieveMulti(opt Options, cmd *cobra.Command, args []string) ([]interface{}, error) {
```
I had to extract this into a separate function to get golangci-linter happy.

nshalman reviewed (commented):
This all looks reasonable to me. Any idea why CI is failing, @mmlb?

jacobweinstock commented:
Hey @mmlb, is this still on your radar? Any help needed?

mmlb commented:
> Hey @mmlb, is this still on your radar? Any help needed?

Still on the radar not sure why I hadn't merged it yet. I think I wanted to give it a go via sandbox but forgot/got distracted with boot issues I was debugging :D. I'll get CI to pass and get it merged soon.

mmlb commented:
> This all looks reasonable to me. Any idea why CI is failing, @mmlb?

Rebased on top of master and now failing on actions/setup-go not being nice and exporting `GOROOT` which messes with the go being run within nix-shell. I'm going to open up a PR to actions/setup-go so it stops doing this.

mmlb commented:
Waiting on https://github.com/actions/setup-go/pull/175

mmlb commented:
I'll rebase on top of main now that https://github.com/tinkerbell/tink/pull/582 should be unblocking this.

jacobweinstock commented:
Hey @mmlb, just a ping on this one.

mmlb commented:
Thanks for the ping @jacobweinstock, lets see how things go.

mmlb commented:
ah the tests are still going to fail though, lets see if thats a quick fix.

mmlb commented:
alright @jacobweinstock I'm all rebased and tests passing!

mmlb commented:
Hey @jacobweinstock ptal again. I added a couple more tests, for when the function that fetches the data doesn't exist. And added some protection to avoid running an undefined function.

Belchy06 commented:
Awesome to see this merged in.

Sorry for going MIA, my studies and work absolutely swamped me and I was unable to set some time aside to get the original PR complete.

Thanks guys!

mmlb commented:
@Belchy06 no worries, thanks for kicking this off.
